<launch>
<!-- Launch file for ouster OS0-128 LiDAR -->

    <include file="$(find fast_lio)/launch/nce_ouster_128_eval.launch">

    </include>
    
    <arg name="num_filter_points" default="4"/>
    <arg name="size_surf" default="0.5"/>
    <arg name="size_map" default="0.5"/>

    <arg name="use_channel" default="0"/>
    <arg name="params" default='{"type": "weighted_da", "w_dist": 0.905372, "w_angle": -0.215389}'/>
    <arg name="model" default="weighted_da"/>
    <!-- arg name="model" default="None"/ -->

	<arg name="data_set" default="nc_math/" />
    <arg name="data_path" value="/home/jhorn/Documents/Uni/MasterThesis/data/" />
    <arg name="folder" value="$(arg data_path)/$(arg data_set)/" />
    
    <arg name="grad_w" default="0.0" />
    <param name="ref_grad_w" type="double" value="$(arg grad_w)"/>
    <param name="filter_size_surf" type="double" value="$(arg size_surf)" />
    <param name="filter_size_map" type="double" value="$(arg size_map)" />
    <param name="point_filter_num" type="int" value="$(arg num_filter_points)"/>
    <param name="filter_use_channel" type="int" value="$(arg use_channel)"/>
    <param name="comp_model" type="string" value="$(arg model)"/>
    <param name="comp_params" type="string" value="$(arg params)"/>
    <param name="w_filter_size" type="int" value="5"/>
    <param name="h_filter_size" type="int" value="4"/>
    <arg name="out_bag" value="$(arg folder)/path_w$(arg grad_w).bag" />
    <!-- node pkg="rosbag" type="record" name="record_traj" args="/Odometry -O $(arg out_bag)" / -->
    
    <arg name="bag_path" value="$(arg folder)/raw_data.bag"/>
    <!-- arg name="bag_path" value="$(arg data_set)"/ -->
    <arg name="bag_args" value="--clock --start=5 --delay=3 --rate=0.5"/>
    <node pkg="rosbag" type="play" name="player" output="screen" args="$(arg bag_args) $(arg bag_path)" required="true"/>

</launch>
